[CORL-1526] Allow sorting moderation queues

nick-funk opened the pull request:
## What does this PR do?

Adds a sort filter to the moderation queues.

## What changes to the GraphQL/Database Schema does this PR introduce?

- Adds `orderBy` props to comments in various places on the moderation queue schema queries

## How do I test this PR?

- Head to admin
- Go to mod queues
- Change the sort order

## TODO
- [x] Use `matchContext` from `found` to get sort order from relay cache instead of `localStorage`
- [x] Implement `initialOrderBy` for approved and rejected queue
- [x] Live updates during sort by oldest should follow stream implementation

nick-funk committed:
Plumb sort order prop through mod queue routes

CORL-1526

nick-funk committed:
Store mod queue sort order in local storage

CORL-1526

nick-funk committed:
Load the sort order from localStorage for now

CORL-1526

nick-funk committed:
Add View New to bottom of mod queue for oldest sort order

CORL-1526

nick-funk committed:
Preliminarily hook up sorted mod queue subscriptions

CORL-1526

nick-funk committed:
Remove unnecessary await after rebase

CORL-1526

nick-funk committed:
Make refresh work for sorted mod queue

CORL-1526

nick-funk committed:
Convert RejectedQueueRoute to FunctionComponent

CORL-1526

nick-funk committed:
Convert ApprovedQueueRoute to FunctionComponent

CORL-1526

nick-funk committed:
Preliminarily hook up approve, rejected sorting

CORL-1526

nick-funk committed:
Fix connection filtering on cached queue data, update tests

CORL-1526

alinhle committed:
feat+chore: upgrade found packages and type matchContext

alinhle committed:
fix: don't use localStorage directly but from local state

alinhle committed:
fix: implement initialOrderBy for approved and rejected

alinhle committed:
fix: add orderBy to cache key in QueueRoute

alinhle committed:
fix: remove unused

alinhle commented on src/core/server/graph/resolvers/ModerationQueue.ts:
```diff
@@ -25,12 +25,17 @@ export const ModerationQueue: GQLModerationQueueTypeResolver<ModerationQueueInpu
+    { first, after, orderBy },
+    { mongo, tenant, logger }
+  ) => {
+    logger.info({ connection, first, after, orderBy }, "WYATT");
```
Is this solved? :-D

alinhle commented on src/core/client/admin/routes/Moderate/Queue/Queue.tsx:
```diff
@@ -154,6 +165,20 @@ const Queue: FunctionComponent<Props> = ({
+        </Flex>
+      )}
+
       {hasMore && (
```
Live updates for `oldest` sort should be implemented like on the comment stream. Instead of showing a button here, we should just tell relay that there is more data to fetch und the ui handles the rest.

See: https://github.com/coralproject/talk/blob/f070cbc0e3ac59f4b3cbafc1474c8a3713b59b5e/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/CommentEnteredSubscription.tsx#L56

Then we can stop the `QueueCommentEnteredSubscription` completely when there is more data. See implementation in stream:
https://github.com/coralproject/talk/blob/f070cbc0e3ac59f4b3cbafc1474c8a3713b59b5e/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabContainer.tsx#L104

alinhle commented on src/core/server/graph/loaders/Comments.ts:
```diff
@@ -217,11 +217,12 @@ export default (ctx: GraphContext) => ({
       first: defaultTo(first, 10),
       after,
-      orderBy: GQLCOMMENT_SORT.CREATED_AT_DESC,
+      orderBy: orderBy ? orderBy : GQLCOMMENT_SORT.CREATED_AT_DESC,
```
Maybe use `defaultTo` as well?

nick-funk committed:
Update queue comment entered subscr to show load more

Instead of showing view new at both top and bottom of queue,
just let the pagination load in more at the bottom.

CORL-1526

alinhle reviewed (changes requested):
Wow, works really nice!

Can you also stop the subscription when there is a next page like on the stream? That would be perfect!

https://github.com/coralproject/talk/blob/f070cbc0e3ac59f4b3cbafc1474c8a3713b59b5e/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabContainer.tsx#L104

Also while you're at it, don't forget the 
`defaultTo` thing: https://github.com/coralproject/talk/pull/3351/files/999f1d4377066adcd0d605c682de6eb166a72795#diff-fa2f72f0134a75b6ab87c8bde733d43241a2718af0de2b8b101f29c39b7aff51

Thanks!

nick-funk committed:
Remove unnecessary logger call

CORL-1526

nick-funk committed:
Replace ternary with defaultTo in Comments loader

CORL-1526

nick-funk committed:
Conditionally subscribe to comment entered/left in queues

CORL-1526

alinhle commented on src/core/client/admin/routes/Moderate/Queue/QueueRoute.tsx:
```diff
@@ -130,6 +151,7 @@ export const QueueRoute: FunctionComponent<Props> = ({
     subscribeToQueueCommentEntered,
     subscribeToQueueCommentLeft,
```
Almost! While this behavior works for `subscribeToQueueCommentEntered `, `subscribeToQueueCommentLeft` must be always on.

nick-funk committed:
Always subscribe to comment left in QueueRoute

CORL-1526

nick-funk committed:
Break queue subscriptions into two effects

CORL-1526

kodiakhq[bot] committed:
Merge branch 'develop' into feat/CORL-1526-mod-queue-sort
